src/lib/components/Sidebar.svelte

```diff
@@ -4,14 +4,88 @@
   import type { UserWithUserTeamsActiveTeamAndChats } from '$lib/server/entities/user'
   import { isSidebarOpen } from '$lib/stores/general'
   import { ChevronRightIcon, PlusIcon } from '@babeard/svelte-heroicons/solid'
+  import {
+    getTimeSince,
+    isToday,
+    isLastYear,
+    isYesterday,
+    millisecondsPerDay,
+    previousSevenDays,
+    lastThirtyDays,
+    currentYear,
+  } from '$lib/utils/time'
 
   export let user: UserWithUserTeamsActiveTeamAndChats
+
+  const renderedTitles: { [key: string]: boolean } = {}
+
   $: chats = [
     ...(user?.sharedChats
       .filter((x) => x.chat.owner.teamId == user.activeUserTeam?.teamId)
       .map((x) => x.chat) || []),
     ...(user.activeUserTeam?.chats || []),
   ].sort((a, b) => b.updatedAt.getTime() - a.updatedAt.getTime())
+
+  type ChatsAccumulator = {
+    [key: string]: any[]
+    today: any[]
+    yesterday: any[]
+    previousSevenDays: any[]
+    lastMonth: any[]
+  }
+
+  $: chatsGroupedByTime = chats?.reduce(
+    (ackumulator: ChatsAccumulator, chat) => {
+      if (isToday(chat.updatedAt)) {
+        //!ackumulator['today'] ? (ackumulator['today'] = []) : ackumulator['today'].push(chat)
```

Comment:
This comment here is because I'm thinking about if this is a better way of writing this or not